src/utils/validateEventData.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ *   Takes the eventData passed in from external use of a tracker and assets all values are of type string
+ *   Excludes statusCode as this value is stringified as part of the DLP
+ */
+
+export default function validateEventData(eventData: Object): boolean {
```

Comment:
In the documentation (https://govukverify.atlassian.net/wiki/spaces/PI/pages/3603791886/WIP+GA4+Implementation+Guide#On-page-load-IN-SCOPE), there is also a limit on the number of characters possible. Do we want to check that also?
And all the fields are mandatory and if no value we have to set "undefined", do we want to check that here or somewhere else?

Reply:
In this function, we are not testing the type of statusCode. That can be an issue.

Reply:
I can extend the function to check for statusCode, it was initially left out as we stringify it as part of the dataLayerPush itself, `pageViewTracker.ts line 23`. For the purposes of the test (and as it should come through as a number) I can check that we're having a number passed in for that rather than a string.

Will extend for character length and that there is a value passed in